### PR TITLE
Use typed and explicitly named names for image fields in containers

### PR DESCRIPTION
**Pull request by mtrmac**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

As a ~side-effect of an audit of how image-related data is produced, stored on disk, and used, rename the fields typically named (`image`, `imageName` and `imageRef`) to (`userRequestedHeuristicImage`, `someNameOfTheContainersImage` and `imageID`), and use strong types for the latter two, throughout the codebase.

As a side effect of this analysis, this allows us to remove one on-disk image lookup on the container creation path.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

See individual commit messages for details; notably this introduces new failure cases on invalid image names in storage, that might not be the best choice.

I’m stopping (after all those changes…) at the point where I’m confident that once we have an `imageID`, it is the only value that matters for running containers (assuming the checkpoint creation was created with a version that sets `RootfsImageRef`); and where it is ~explicit whether various APIs return the `userRequestedHeuristicImage` or the `someNameOfTheContainersImage` value.

AFAICT Kubelet never uses the image name values (only `ImageRef` = image ID), they “only”, show up in Pod statuses, so that’s where relevance to image signing goes.

Arguably the PR should do _something_ about the `userRequestedHeuristicImage` / `someNameOfTheContainersImage` values: maybe only ever use one of them, quite possibly make some effort to make `someNameOfTheContainersImage` more related to users’ input. If you think that’s required, please say so (and preferably choose a design).

Cc: @saschagrunert 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: server/container_start.go
Comment (mtrmac) [inline server/container_start.go:34]: I don’t understand what is really going on here…

File: server/mock_helpers_test.go
Comment (mtrmac) [inline server/mock_helpers_test.go:15]: This is a copy of code from `internal/storage`; is there a good place to share it?

File: server/server.go
Comment (mtrmac) [inline server/server.go:220]: Looking at coverage data… apparently the existing `"should succeed with container restore"` test exercises this using 3 containers with `ID` and `ImageID` both empty. I’m… getting tired of fixing mocks that just don’t represent real-world operation.

**Review by saschagrunert — APPROVED** [review]

File: server/mock_helpers_test.go
Comment (saschagrunert) [reply server/mock_helpers_test.go:15]: How about creating a new package in `internal` for this?

File: internal/oci/container.go
Comment (haircommander) [inline internal/oci/container.go:62]: personally, I think continuing to refer to it as `imageName` and commenting the caveats is sufficient.

File: internal/factory/container/container.go
Comment (haircommander) [inline internal/factory/container/container.go:76]: similarly here. While i can see a case for being clearer, I think the variable name is very verbose and could be shortened

Comment (haircommander) [reply internal/factory/container/container.go:76]: maybe `UserRequestedImage` ?

Comment (mtrmac) [reply internal/factory/container/container.go:76]: My thinking was that scary concepts should have scary names, to discourage future readers from adding any more uses… and even perhaps motivate (… someone else …) to replace these values entirely.

But really 90% of the value of the exercise for me is in having found/tracked those values, so I’m happy with any names the maintainers prefer.

File: server/container_start.go
Comment (mtrmac) [reply server/container_start.go]: … anyway, I have outright removed that field value. It clearly doesn’t do anything, so let’s not even pretend to.

File: server/mock_helpers_test.go
Comment (mtrmac) [reply server/mock_helpers_test.go]: Moved to `internal/mockutils`.

File: internal/oci/container.go
Comment (mtrmac) [reply internal/oci/container.go]: Reverted (but kept the explicit type).

File: internal/factory/container/container.go
Comment (mtrmac) [reply internal/factory/container/container.go]: Changed to `UserRequestedImage`.

File: server/server.go
Comment (mtrmac) [reply server/server.go:220]: I have changed that to at least use 3 proper, and distinct, values. Not that it exercises the wipe logic all that much…

**Review by saschagrunert — APPROVED** [review]
/hold

for @haircommander discussions

LGTM from my side

**Pull request merged**
